R/test-app.R

```diff
@@ -99,6 +81,74 @@ testApp <- function(appDir = ".", testnames = NULL, quiet = FALSE,
   )
 }
 
+#' Identify in which directory the tests are contained.
+#'
+#' Prior to 1.3.1.9999, tests were stored directly in `tests/` rather than
+#' nested in `tests/shinytests/`.
+#'
+#' This function does the following:
+#'  1. Check to see if `tests/shinytests/` exists. If so, use it.
+#'  2. Check to see if all the top-level R files in `tests/` appear to be shinytests. If
+#'     some are and some aren't, throw an error.
+#'  3. Assuming all top-level R files in `tests/` appear to be shinytests, return that dir.
+#'  @noRd
+findTestsDir <- function(appDir) {
+  testsDir <- file.path(appDir, "tests")
+
+  if (!dir.exists(testsDir)){
```

Comment:
The `dir.exists` function was added to R in version 3.2.0. Because older versions of R didn't have `dir.exists`, shinytest has a `dir_exists` function that does the same thing.

Although it's probably OK to start using `dir.exists` at this point, to be extra safe, we should keep using `dir_exists` for now.

Reply:
Ah, good to know!